src/Form.js

```diff
@@ -5,40 +5,43 @@ function Form({ pokemon, setPokemon }) {
     const [searchTerm, setSearchTerm] = useState("");
     const [allPokemon, setAllPokemon] = useState([]);
 
-    function titleCase(string) {
-        const firstLetter = string.charAt(0).toUpperCase();
-        const restOfTheWord = string.slice(1).toLowerCase();
-        return firstLetter + restOfTheWord;
-    }
+    // When the component loads initially,
+    // make a copy of the passed in pokemon array
+    // if and only if the copy of the array is empty.
+    useEffect(() => {
+        if (allPokemon.length === 0) {
+            setAllPokemon(pokemon);
+        }
+    }, [pokemon, allPokemon]);
```

Comment:
Most of my suggestions were in the Form component. Be sure to review the diffs in the "Files changed" tab to see all of the changes that I made along with the inline code comment explanations.